front/app/containers/UsersEditPage/ProfileForm.tsx

```diff
@@ -85,259 +81,175 @@ interface DataProps {
-
-    const newValues = Object.entries(this.state.extraFormData).reduce(
-      (acc, [key, extraFormDataConfiguration]) => ({
-        ...acc,
-        [key]: extraFormDataConfiguration?.formData,
-      }),
-      values
+  }, [authUser, methods]);
+
+  if (isNilOrError(authUser)) return null;
+
+  const onFormSubmit = async (formValues: FormValues) => {
+    const avatar = formValues.avatar ? formValues.avatar[0].base64 : null;
+    // Add custom fields values to form
+    const newFormValues = Object.entries(extraFormData).reduce(
+      (acc, [key, extraFormDataConfiguration]) => {
+        return {
+          ...acc,
+          [key]: extraFormDataConfiguration?.formData,
+        };
+      },
+      formValues
     );
 
-    setStatus('');
+    eventEmitter.emit('customFieldsSubmitEvent');
+    Object.values(extraFormData).forEach((configuration) => {
+      configuration?.submit?.();
+    });
 
     try {
-      await updateUser(authUser.id, newValues);
+      await updateUser(authUser.id, { ...newFormValues, avatar });
       streams.fetchAllWith({
         apiEndpoint: [`${API_PATH}/onboarding_campaigns/current`],
       });
-      resetForm();
-      setStatus('success');
-    } catch (errorResponse) {
-      if (isCLErrorJSON(errorResponse)) {
-        const apiErrors = (errorResponse as CLErrorsJSON).json.errors;
-        setErrors(apiErrors);
-      } else {
-        setStatus('error');
-      }
-      setSubmitting(false);
+    } catch (error) {
+      handleHookFormSubmissionError(error, methods.setError);
     }
   };
 
-  formikRender = (props) => {
-    const {
-      values,
-      errors,
-      setFieldValue,
-      setFieldTouched,
-      setStatus,
-      isSubmitting,
-      submitForm,
-      isValid,
-      status,
-      touched,
-    } = props;
-    const { lockedFields, authUser, disableBio } = this.props;
-    const { hasPasswordMinimumLengthError } = this.state;
-
-    // Won't be called with a nil or error user.
-    if (isNilOrError(authUser)) return null;
-
-    const lockedFieldsNames = isNilOrError(lockedFields)
-      ? []
-      : lockedFields.map((field) => field.attributes.name);
-
-    const { formatMessage } = this.props.intl;
-
-    const isExtraFormDataTouched = () =>
-      Object.values(this.state.extraFormData).some(
-        (value) => !isEmpty(value) && Object.keys(value ?? {}).length > 1
-      );
-
-    const getStatus = () => {
-      let returnValue: 'enabled' | 'disabled' | 'error' | 'success' = 'enabled';
-
-      if (isSubmitting) {
-        returnValue = 'disabled';
-      } else if (
-        (!isEmpty(touched) && !isValid && !isExtraFormDataTouched()) ||
-        status === 'error'
-      ) {
-        returnValue = 'error';
-      } else if (isEmpty(touched) && status === 'success') {
-        returnValue = 'success';
-      }
+  const lockedFieldsNames = isNilOrError(lockedFields)
+    ? []
+    : lockedFields.map((field) => field.attributes.name);
+
+  const handleFormOnChange = ({
```

Comment:
Was confused about this at first, could benefit from changing the name to e.g. `handleCustomFieldsChange` or something referring to that subform